src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -283,10 +281,9 @@ public bool ContainsValue(TValue value)
             }
             else
             {
-                EqualityComparer<TValue> c = EqualityComparer<TValue>.Default;
                 for (int i = 0; i < count; i++)
                 {
-                    if (entries[i].hashCode >= 0 && c.Equals(entries[i].value, value)) return true;
+                    if (entries[i].hashCode >= 0 && (default(TValue) == null ? value.Equals(entries[i].value) : EqualityComparer<TValue>.Default.Equals(entries[i].value, value))) return true;
```

Comment:
This is functional change, e.g. when the value implements IEquatable.

Reply:
fixed